src/main/java/com/google/cloud/spanner/pgadapter/statements/IntermediateStatement.java

```diff
@@ -192,9 +242,17 @@ protected void handleExecutionException(SQLException e) {
   /** Execute the SQL statement, storing metadata. */
   public void execute() {
     this.executed = true;
+    int[] updateCounts = null;
     try {
-      this.statement.execute(this.sql);
-      this.executeHelper();
+      if (statements.size() > 1) {
+        for (String stmt : statements) {
+          this.statement.addBatch(stmt);
+        }
+        updateCounts = this.statement.executeBatch();
```

Comment:
can this return an exception ? example, what if there is a commit message in here

Reply:
Both 'addBatch' and 'executeBatch' can return exceptions.  This is inside of a try/catch block.

Reply:
Its important to think of error responses and scenarios. A meaningful error message will help the customer, example: if it only says InvalidArg but its not clear to the customer since PGAdapter supports COMMIT as a single statement but fails in -c option now.

Reply:
The error is not vague.  For BEGIN and COMMIT it will produce this error:
"ERROR:  INVALID_ARGUMENT: The statement is not suitable for batching. Only DML and DDL statements are allowed for batching."

Reply:
discussed adding a section in README to provide all this context and info